tests/unit/test_sonata_config.py

```diff
@@ -1,146 +1,106 @@
+        }
+    },
+], indirect=True)
+def test_parse_run(create_tmp_simulation_config_file):
     from neurodamus.core.configuration import SimConfig
-    SimConfig.init(SONATA_CONF_FILE, {})
+    SimConfig.init(create_tmp_simulation_config_file, {})
     # RNGSettings in hoc correctly initialized from Sonata
     assert SimConfig.rng_info.getGlobalSeed() == 1122
 
-    # run section
-    assert SimConfig.run_conf['CircuitTarget'] == 'Mosaic'
-    assert SimConfig.run_conf['Simulator'] == 'NEURON'
-    assert SimConfig.run_conf['Duration'] == 50.0
-    assert SimConfig.run_conf['Dt'] == 0.1
-    assert SimConfig.run_conf['Celsius'] == 35
-    assert SimConfig.run_conf['V_Init'] == -75
-    assert SimConfig.run_conf['SpikeLocation'] == 'soma'
+    run_conf = SimConfig.run_conf
+    assert run_conf['CircuitTarget'] == 'Mosaic'
+    assert run_conf['Simulator'] == 'NEURON'
+    assert run_conf['Duration'] == 50.0
+    assert run_conf['Dt'] == 0.1
+    assert run_conf['Celsius'] == 35
+    assert run_conf['V_Init'] == -65
+    assert run_conf['SpikeLocation'] == 'soma'
+    assert run_conf['ExtracellularCalcium'] == 1.2
 
-    # output section
-    assert SimConfig.run_conf['SpikesFile'] == 'spikes.h5'
-    assert SimConfig.run_conf['SpikesSortOrder'] == 'by_time'
 
-    # reports section
-    soma_report = SimConfig.reports['soma_report']
-    assert soma_report['Target'] == 'l4pc'
-    assert soma_report['Type'] == 'compartment'
-    assert soma_report['ReportOn'] == 'v'
-    assert soma_report['Compartments'] == 'center'
-    assert soma_report['Sections'] == 'soma'
-    assert soma_report['Scaling'] == 'Area'
-    assert soma_report['StartTime'] == 0.0
-    assert soma_report['EndTime'] == 50.0
-    assert soma_report['Dt'] == 0.1
-    assert soma_report['Enabled']
-    compartment_report = SimConfig.reports['compartment_report']
-    assert soma_report['FileName'] == str(Path(SimConfig.run_conf["OutputRoot"]) / "soma_report.h5")
-    assert compartment_report['Target'] == 'l4pc'
-    assert compartment_report['Type'] == 'compartment'
-    assert compartment_report['ReportOn'] == 'v'
-    assert compartment_report['Compartments'] == 'all'
-    assert compartment_report['Sections'] == 'all'
-    assert compartment_report['Scaling'] == 'Area'
-    assert compartment_report['StartTime'] == 0.0
-    assert compartment_report['EndTime'] == 10.0
-    assert compartment_report['Dt'] == 0.1
-    assert compartment_report['Enabled']
-    assert compartment_report['FileName'] == str(
-        Path(SimConfig.run_conf["OutputRoot"]) / "my_compartment_report.h5"
-        )
+@pytest.mark.parametrize("create_tmp_simulation_config_file", [
+    {
+        "simconfig_fixture": "ringtest_baseconfig",
+        "extra_config": {
+            "conditions": {
+                "extracellular_calcium": 1.2,
+                "mechanisms": {
+                    "GluSynapse": {
+                        "cao_CR": 2.0,
+                    }
+                }
```

Comment:
if the default value of cao_CR is 2.0 you do not need this. If it is not, you should still check that the value is 2.0 and is there. An old bug I solved was that every entry in extra_config was replacing, not merging with the default config.

Reply:
BTW, Neurodamus doesn't check the default value of the global variable in the mechansim rather than applying the "new" value directly. It is up to the user to decide which variable and its value to include in the "conditions/mechanisms".